enhancements/management-workload-partitioning-api.md

```diff
@@ -0,0 +1,239 @@
+
+The short term goal is to get enough functionality to create the `management` partition where all OpenShift infrastructure could be placed. This partition is cluster wide and only affects pods with certain pre-agreed-upon labels and placed in specific pre-annotated namespaces.
+
+### Non-Goals
+
+- Long term goal of allowing cluster admins to create their own partitions is not part of this proposal for now
+- Day 2 creation of partitions is not part of the proposal
+- Namespace or MCP specific partitions are not part of the proposal
+
+## Proposal
+
+The proposal is to define a new cluster-wide Custom Resource Definition that would describe the allowed partition names in the status section. That way it hints at being a read only object where no user/admin input or modifications are expected.
+
+```yaml
+apiVersion: workloadpartitioning.openshift.io/v1
+kind: WorkloadPartitions
+metadata:
+  # arbitrary name, all objects of this Kind should be processed and merged
+  name: management-partition
+status:
+  # list of strings, defines partition names to be exposed by kubelet
+  globalPartitionNames:
+    - management
+```
+
+To allow for future extensibility and possible multiple sources of workload partition names (coming from customers, the installer, or other operators, etc.), we propose that there might be multiple `WorkloadPartitions` objects injected into the cluster. The expected behavior is that all components would just merge all the defined names together.
+
+Only the cluster administrator will have the ability to create or manipulate the WorkloadPartitions objects. Anyone will be allowed to read them.
+
+### User Stories
+
+Please check the user stories in the parent [Workload partitioning management](management-workload-partitioning.md) feature.
+
+The gist of this piece is that it is expected to define the workload partitions before the installation of a cluster starts. A cluster with no `WorkloadPartitions` objects present at the installation time will not have workload partitioning enabled.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+1) An admin creates a WorkloadPartitions object manually after the cluster has been running for some time on a cluster that already has partitioning enabled
+1) An admin creates the WorkloadPartitions object manually after the cluster has been running for some time on a cluster with no workload partitioning enabled
+1) An admin deletes the WorkloadPartitions object that was created during the install process
+1) A random user manages to create a WorkloadPartitions object due to a bug in the defined RBAC rules
```

Comment:
This list is a good start. We should add some detail about what ill effects might result from each of these cases.